src/Bridges/Constraint/bridges/functionize.jl

```diff
@@ -236,188 +263,144 @@ function MOI.supports_constraint(
     return MOI.Utilities.convert_approx(MOI.VectorOfVariables, f)
 end
 
 """
-    ScalarQuadraticToScalarNonlinearBridge{T,S} <: Bridges.Constraint.AbstractBridge
+    FunctionConversionBridge{T,F,G,S} <: AbstractFunctionConversionBridge{G,S}
 
-`ScalarQuadraticToScalarNonlinearBridge` implements the following reformulations:
+`FunctionConversionBridge` implements the following reformulations:
 
   * ``f(x) \\in S`` into ``g(x) \\in S``
 
-where `f` is a [`MOI.ScalarQuadraticFunction`](@ref) and `g` is a
-[`MOI.ScalarNonlinearFunction{T}`](@ref).
+for these pairs of functions:
+
+ * [`MOI.ScalarAffineFunction`](@ref)` to [`MOI.ScalarQuadraticFunction`](@ref)
+ * [`MOI.ScalarQuadraticFunction`](@ref)  to [`MOI.ScalarNonlinearFunction`](@ref)
+ * [`MOI.VectorAffineFunction`](@ref) to [`MOI.VectorQuadraticFunction`](@ref)
 
 ## Source node
 
-`ScalarQuadraticToScalarNonlinearBridge` supports:
+`FunctionConversionBridge` supports:
 
-  * [`MOI.ScalarQuadraticFunction`](@ref) in `S`
+  * `F` in `S`
 
 ## Target nodes
 
-`ScalarQuadraticToScalarNonlinearBridge` creates:
+`FunctionConversionBridge` creates:
 
-  * [`MOI.ScalarNonlinearFunction{T}`](@ref) in `S`
+  * `G` in `S`
 """
-struct ScalarQuadraticToScalarNonlinearBridge{T,S} <:
-       AbstractFunctionConversionBridge{MOI.ScalarNonlinearFunction,S}
-    constraint::MOI.ConstraintIndex{MOI.ScalarNonlinearFunction,S}
+struct FunctionConversionBridge{T,F,G,S} <:
+       AbstractFunctionConversionBridge{G,S}
+    constraint::MOI.ConstraintIndex{G,S}
 end
 
-const ScalarQuadraticToScalarNonlinear{T,OT<:MOI.ModelLike} =
-    SingleBridgeOptimizer{ScalarQuadraticToScalarNonlinearBridge{T},OT}
+const FunctionConversion{T,OT<:MOI.ModelLike} =
+    SingleBridgeOptimizer{FunctionConversionBridge{T},OT}
 
 function bridge_constraint(
-    ::Type{ScalarQuadraticToScalarNonlinearBridge{T,S}},
+    ::Type{FunctionConversionBridge{T,F,G,S}},
     model::MOI.ModelLike,
-    f::MOI.ScalarQuadraticFunction{T},
+    f::F,
     s::S,
-) where {T,S}
-    ci = MOI.add_constraint(model, convert(MOI.ScalarNonlinearFunction, f), s)
-    return ScalarQuadraticToScalarNonlinearBridge{T,S}(ci)
+) where {T,F,G,S}
+    ci = MOI.add_constraint(model, convert(G, f), s)
+    return FunctionConversionBridge{T,F,G,S}(ci)
 end
 
+function MOI.get(
+    model::MOI.ModelLike,
+    ::MOI.ConstraintFunction,
+    b::FunctionConversionBridge{T,F},
+) where {T,F}
+    g = MOI.get(model, MOI.ConstraintFunction(), b.constraint)
+    return convert(F, g)
+end
+
+# VariableIndex -> ScalarAffineFunction  # Handled by ScalarFunctionizeBridge
+
+# ScalarAffineFunction -> ScalarQuadraticFunction
+
 function MOI.supports_constraint(
-    ::Type{ScalarQuadraticToScalarNonlinearBridge{T}},
-    ::Type{MOI.ScalarQuadraticFunction{T}},
+    ::Type{<:FunctionConversionBridge{T}},
+    ::Type{MOI.ScalarAffineFunction{T}},
     ::Type{<:MOI.AbstractScalarSet},
 ) where {T}
     return true
 end
```

Comment:
@blegat, functions for this bridge are opt-in, you need to explicitly implement `supports_constraint` and `concrete_bridge_type`.